merino-web/src/endpoints/suggest.rs

```diff
@@ -169,3 +174,50 @@ impl<'a> Serialize for SuggestionWrapper<'a> {
         generated.serialize(serializer)
     }
 }
+
+/// Log a suggestion request, respecting the log_query setting passed, and
+/// formatting all fields in a way that is helpful to our downstream log
+/// handlers. This primarily means that fields don't generate internal
+/// stringified JSON.
+fn safe_log_request(
+    log_query: bool,
+    request: &SuggestionRequest,
+    query_params: &SuggestQueryParameters,
+) {
+    let none = "none".to_string();
+    let country = request.country.as_ref().unwrap_or(&none);
+    let region = request.region.as_ref().unwrap_or(&none);
+    let city = request.city.as_ref().unwrap_or(&none);
+    let dma = request.dma.map_or_else(|| none.clone(), |v| v.to_string());
+    let query = if log_query {
+        request.query.as_str()
+    } else {
+        ""
+    };
+    let requested_providers = query_params.providers.as_ref().map_or_else(
+        || "".to_string(),
+        |ps| {
+            ps.iter()
+                .map(|p| p.as_str())
+                .collect::<Vec<_>>()
+                .as_slice()
+                .join(",")
+        },
+    );
+
+    tracing::info!(
```

Comment:
I don't see request ID included in this message.

Reply:
It's not included directly in this message. One property of the tracing library is that log messages inherit context from their surrounding. One of those bits of context is the MozLog middlewhare that's [used for all requests](https://github.com/mozilla-services/merino/blob/5876e0346d31b615b832fe8ec0c1620b7b7efff4/merino-web/src/lib.rs#L120). It adds things like `rid` to all logging that happens in a request, so that they can be correlated together.

Reply:
Can we add a comment here that mentions this fact? I think there's value to being able to point to this set of lines and be audit exactly what output Merino is producing for this message type.

Reply:
Instead of manual auditing, how does it sound to add that rid must be included to the test for this logging?

Reply:
> Instead of manual auditing, how does it sound to add that rid must be included to the test for this logging?

My expectation is that if I'm browsing this codebase and I see this section where we're logging the query string, I want to be able to quickly discover the context to be able to verify what all is being sent here. So if we put this responsibility on testing, I would want to make sure we add a comment before this block that indicates to the reader that the test validates the contents.

If we do this in testing, just checking for existence of `rid` doesn't seem sufficient. I'd want to test that there aren't additional fields we weren't expecting. If the number of contextual fields is likely to stay small, then codifying it in a test might be maintainable.

Reply:
I've added some comments. Also note that all of this is documented for a reason. Ideally one shouldn't need to be digging through the code to find all this.

Reply:
> Instead of manual auditing, how does it sound to add that rid must be included to the test for this logging?

PS: Turns out that this is very hard to test because of mixing semantic and formatting concerns in tracing-actix-web-mozlog. I filed https://github.com/mozilla-services/common-rs/issues/27 for that.